osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeConverter.java

```diff
@@ -66,13 +76,11 @@
                     case POINT: {
                         PointShape aPoint = (PointShape) s;
                         Marker m = new Marker(map);
-                        m.setPosition(new GeoPoint(aPoint.getY(), aPoint.getX()));
-                        if (metadata != null) {
-                            metadata.setStringCharset(Charset.defaultCharset());
-                            m.setSnippet(metadata.toMap().toString());
-                            m.setTitle(getSensibleTitle(m.getSnippet()));
+                        m.setPosition(fixOutOfRange(new GeoPoint(aPoint.getY(), aPoint.getX())));
 
-                        }
+                        shapeMetaSetter.set(metadata, m);
```

Comment:
Just `shapeMetaSetter.set(metadata, m)`: the other two lines `m.setSnippet(shapeMetaSetter.getSnippet())` and `m.setTitle(shapeMetaSetter.getTitle())` will be performed inside `shapeMetaSetter.set(metadata, m)`.